working/extension_structs/overview.md

```diff
@@ -0,0 +1,815 @@
+
+```dart
+struct Data();
+struct GenericData<T>();
+```
+
+#### Alternatives
+
+
+##### Re-use the class keyword
+
+An alternative is to continue to use "class" and to use some other piece of
+syntax to indicate that the object in question is not a general class.  For
+example, the parenthesized list might be enough:
+
+
+```dart
+class Data();
+class GenericData<T>();
+```
+
+Alternatively, an additional keyword could be used:
+
+```dart
+class Data wraps ();
+class GenericData<T> wraps ();
+```
+
+Other keywords could be considered.
+
+##### Choose a different leading keyword
+
+Instead of "struct", we could use "view", or some other choice:
+
+```dart
+view Data();
+view GenericData<T>();
+```
+
+##### Use a modifier on classes
+
+
+We could use "data class":
+
+```dart
+data class Data();
+data class GenericData<T>();
+```
+
+Another option would be "view class".
+
+
+### Fields and primary constructors
+
+The only new piece of syntax in this section of the proposal (other than the
+"struct" keyword) is the primary constructor which follows immediately after the
+class name + generics.  The primary constructor serves both to define the
+fields, and to define the signature of the default generated constructor.
+
+The primary constructor consists of a parenthesized list of comma separated
+`<type> <identifier> (= <expression>)?` entries.  That is, a list of variable
+declarations: with no modifiers; with types required; and with optional
+initializer values.
+
+Each entry in the list is a field in the struct.  Every field is implicitly
+final.  They may not be late, and a type must be provided.
+
+If an initializer is provided for an entry, all subsequent entries must also
+have an initializer provided (see generated members below).
+
+**COMMENTARY(leafp):** *This is to allow initializers to double as default
+  values for the generated constructor.  This may be too cute.  Alternatively we
+  could forbid initializers, or just make fields with initializers not available
+  to the constructor (as with a normal class)*
+
+It is an error if two entries have the same name, and it is an error if one
+entry consists of the name of another entry except prefixed with `_`.  That is,
+all entries must be uniquely named after ignoring privacy.
+
+**COMMENTARY(leafp):** *This is to allow us to use the non-private version of
+  the name as a named parameter in methods*
+
+Examples:
+
+```dart
+struct Data(int x, List<int> l = [3]);
+struct GenericData<T>(T x, T y);
+```
+
+Ignoring generated members, these two structs are roughly equivalent to the
+following classes:
+
+```dart
+class Data {
+  int x;
+  List<int> l = [3]
+ }
+class GenericData<T> {
+ T x;
+ T y;
+ }
+
+```
+
+#### Alternative: Split into positional and named
+
+We could choose to make this look more directly like a constructor argument list
+by allowing "named" parameters inside of a brace delimited set.  These could
+then become named parameters in the default constructor.
```

Comment:
Yeah, I was expecting this when I read the introductory example, that I might have written as:

```dart
struct Component(int x, int y, int z, {int w = 1});

void test() {
  // Call the default constructor, using the default value for w
  var c1 = Component(0, 0, 0);
  // Call the default constructor, explicitly passing w
  var c2 = Component(1, 1, 1, w: 1);
...
}
```

Reply:
Added your example to this section.